hack/gen_old_scheme.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+DIR="$(dirname "${BASH_SOURCE[0]}")"
+ROOT_DIR="$(realpath "${DIR}/..")"
+
+function find_package() {
+  local dir=$1
+  find "${dir}" | grep register.go | sed "s#/register.go##g" | sed "s#${dir}/##g" || :
+}
+
+function gen() {
+  cat <<EOF
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+// Don't edit this file directly. It is generated by gen_old_scheme.sh.
+import (
+EOF
+
+  for pkg in $(find_package "${ROOT_DIR}/pkg/old/apis"); do
+    echo "github.com/etcd-io/auger/pkg/old/apis/${pkg}" | awk -F '/' '{print "	"$7$8, "\""$1"\/"$2"\/"$3"\/"$4"\/"$5"\/"$6"\/"$7"\/"$8"\""}'
```

Comment:
I find shell scripts generally not flexible. There could be api files not necessarily follow this exact pattern. I think go scripts with more string manipulation function would be more flexible. WDYT?

Reply:
Yes, I changed it a little, it looks a little better but not much, this hard code can't be avoided. 😄